AutomatedTesting/Gem/PythonTests/Atom/tests/MaterialEditor_Atom_PeriodicTests.py

```diff
@@ -0,0 +1,409 @@
+            original_material_document_saved is True)
+
+        # 4. Change the baseColor.color property value of original_material_document to 0.5, 0.5, 0.5, 1.0.
+        new_material_document_color = azlmbr.math.Color(0.5, 0.5, 0.5, 1.0)
+        material_editor_utils.set_property(
+            original_material_document_id, base_color_property_name, new_material_document_color)
+        Report.result(
+            Tests.changed_new_material_document_color,
+            material_editor_utils.get_property(
+                original_material_document_id, base_color_property_name) == new_material_document_color)
+
+        # 5. Save a new material document from original_material_document referred to as "new_material_document".
+        new_material_document_saved = atom_tools_utils.save_document_as_copy(
+            original_material_document_id, new_material_document)
+        Report.result(
+            Tests.saved_new_material_document,
+            new_material_document_saved is True)
+
+        # 6. Change the baseColor.color property value of original_material_document to 0.75, 0.75, 0.75, 1.0.
+        child_material_document_color = azlmbr.math.Color(0.75, 0.75, 0.75, 1.0)
+        material_editor_utils.set_property(
+            original_material_document_id, base_color_property_name, child_material_document_color)
+        Report.result(
+            Tests.changed_child_material_document_color,
+            material_editor_utils.get_property(
+                original_material_document_id, base_color_property_name) == child_material_document_color)
+
+        # 7. Save a new child material document referred to as "child_material_document".
+        child_material_document_saved = atom_tools_utils.save_document_as_child(
+            original_material_document_id, child_material_document)
+        Report.result(
+            Tests.saved_child_material_document,
+            child_material_document_saved is True)
+
+        # 8. Close & open previously saved original_material_document, new_material_document, & child_material_document.
+        closed_all_material_documents = atom_tools_utils.close_all_documents()
+        Report.result(
+            Tests.closed_original_new_and_child_material_documents,
+            closed_all_material_documents is True)
+        original_material_document_id = atom_tools_utils.open_document(original_material_document)
+        new_material_document_id = atom_tools_utils.open_document(new_material_document)
+        child_material_document_id = atom_tools_utils.open_document(child_material_document)
+        material_document_ids = [original_material_document_id, new_material_document_id, child_material_document_id]
+        for material_document_id in material_document_ids:
+            Report.result(
+                Tests.opened_original_new_child_material_documents,
+                atom_tools_utils.is_document_open(material_document_id) is True)
```

Comment:
this Report.result runs 3 times but if any one of them fails we get the general all of them failed. it will say `"P0: Failed to open original_material_document, new_material_document, & child_material_document."` should we put an f-string tuple in here that indicaes which material_document_id failed?

Reply:
I am adding f strings inside the for loops